### ISSUE TITLE
[BUG] `split` doesn't handle empty split points

### ISSUE THREAD

**Issue by jrhemstad** [rapidsai/cudf#4656]
**Describe the bug**

`split` doesn't correctly handle an empty `split_points` vector.

**Expected behavior**

If I want to perform an "identity" split operation, I'd pass in an empty `split_points` vector, which should just return a vector with a single `table_view`.

**Comment by jrhemstad** [comment]
Here's the problem:
https://github.com/rapidsai/cudf/blob/branch-0.13/cpp/src/copying/split.cpp#L34

If `splits` is empty, it returns an empty `vector<table_view>`.

Instead, it should do:
```
        if(splits.size() == 0 or column_size == 0) {
            return std::vector<table_view>{input};
        }
```

**Comment by galipremsagar** [comment]
Fixed by: #4660
